src/assumptions.js

```diff
@@ -0,0 +1,94 @@
+// # Checking Data Assumptions
+//
+// Data processing is tricky business, full of pitfalls and gotchas. Hopefully the tasks in this guide help with getting started in this process. But you, I, and the entire world will make mistakes. Its natural.
+//
+// But mistakes in data processing, like all other kinds of mistakes, can be painful. They can result in hours of bug hunting, days of reprocessing, and months of crying. Since we know mistakes happen and will continue to happen, what can we do to take away some of the pain?
+//
+// In a word, _padding_. We need some padding to protect us from the bumps and bruises of data processing. And I would suggest that this padding come in the form of simple tests that check the assumptions you have about the shape and contents of your data. 
+//
+// Unless there is an extreme performance need, these tests should run in the data processing pipeline. Optimally, they would be easy to turn on and off so that you can disable them if you need to if your code is deployed.
+//
+// ## Assertions
+//
+// These tests can be created with [assertions](http://en.wikipedia.org/wiki/Assertion_%28software_development%29) - functions that check the truthiness of a statement in code. Typically, they raise an error when an expected truth is not actually true.
+//
+// JavaScript doesn't have a built assertions, but we can rectify this deficiency with a simple function.
+
+function assert(isTrue, message) {
+  if(!isTrue) {
+    console.log(message);
+    return false;
+  }
+  return true;
+}
+
+// This will output a given message if the input is not true. Typically assertions [throw](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/throw) errors, but we can just log it for explaining purposes.
+//
+// ## Data Content Assumptions
+//
+// Now let's use our `assert` function to check some assumptions about the details of our data.
+//
+// We can use lodash's suite of [type checking functions(https://lodash.com/docs#isBoolean) to take care of performing the checks, passing the result of the check to `assert` to produce our errors.
+//
+// Let's say our data importing process has made some mistakes:
+//
+var data = [{"name":"Dan",
+             "age":23,
+             "superhuman":false},
+            {"name":"Sleepwalker",
+              "age":NaN,
+              "superhuman":"TRUE"}
+];
+
+// Our first entry looks ok, where our second entry has some problems. The age parsing for the immortal [Sleepwalker](http://en.wikipedia.org/wiki/Sleepwalker_%28comics%29) has left him with no age. Also, bad input data has left us with a string in `superhuman`, where we expect a boolean. 
+//
+// A simple assumption checking function that could be run on this data could look something like this:
+
+function checkDataContent(data) {
+  data.forEach(function(d) {
+    var dString = JSON.stringify(d);
+    assert(_.isString(d.name), dString + " has a bad name ");
```

Comment:
Does it make sense to make the messages indicate what the result should be? like "has a bad name. Should be a string"